Orange/widgets/data/owcolor.py

```diff
@@ -350,6 +350,7 @@ def set_data(self, data):
                                         self._create_proxies(domain.metas))
             self.openContext(data)
             self.data = data.transform(domain)
+            self.data.domain = domain
```

Comment:
The first problem sounds like a broken hash on Variable.

As for the fix, I would rather move the data.domain = domain to `transform`method than have every caller do this explicitly.